resources/Extra Materials/Trackforce_DB_full.sql

```diff
@@ -4897,7 +4897,31 @@ insert into admin.tf_interview values (51, 'Easy', 'Adventure', TO_TIMESTAMP('20
 insert into admin.tf_interview values (52, 'Grow', 'Bounty', TO_TIMESTAMP('2017-06-09 12:39:00', 'YYYY-MM-DD HH24:MI:SS'), TO_TIMESTAMP('2017-11-03 17:00:00', 'YYYY-MM-DD HH24:MI:SS'), 'No Reason', TO_TIMESTAMP('2017-08-04 01:00:00', 'YYYY-MM-DD HH24:MI:SS'), 0, 0, 'Programmer', 'Questions...', 1, 150, 577, 1244, 1);
 insert into admin.tf_interview values (53, 'Powerful', 'frighten', TO_TIMESTAMP('2017-08-02 21:00:00', 'YYYY-MM-DD HH24:MI:SS'), TO_TIMESTAMP('2017-05-22 19:00:00', 'YYYY-MM-DD HH24:MI:SS'), 'No Reason', TO_TIMESTAMP('2017-11-01 16:00:00', 'YYYY-MM-DD HH24:MI:SS'), 1, 0, 'Developer', 'Questions...', 1, 714, 577, 1251, 3);
 
-
+CREATE OR REPLACE PROCEDURE admin.delete_old_associates(res OUT NUMBER)
+AS
+    CURSOR c1 IS SELECT u.tf_user_id FROM admin.tf_user u;
+    TYPE id_typ IS TABLE OF c1%ROWTYPE;
+    all_id id_typ;
+BEGIN
+    res := 0;
```

Comment:
Throughout this procedure, it looks like you are using tf_client_start_date as if it referred to the date that this user was registered, but it doesn't seem to me that this is what this column stores. I'm guessing it is supposed to hold the date that an associate starts at a client. In any case, I registered a new associate, and this column is null for that new associate.